### PR TITLE
Add: site generation from documentation

### PR DESCRIPTION
**Pull request by e-pettersson-ericsson**
### Applicable Issues
We need a way to publish our existing documentation to github pages.

### Description of the Change
* Add maven site plugin to pom file. This plugin handles generation of markdown files to .html files.
* All generated files are put in the docs/ directory and then Github automatically publishes these files within 1-2 minutes to github pages. The repository settings needs to be updated to enable generation of github pages from branch master, docs/ directory before merging this pull request.
* .md files inside the wiki directory has been moved to a subdirectory to align with the maven site plugin, causing README links to be updated
* .md files have been updated by removing \<details> \<summary> and \<pre> tags and instead use indented text for code blocks.
* site.xml was added to configure layout of the published site.
* resource plugin was added to pom file. wiki/images/ directory are copied to wiki/resources/images/ directory during execution of site phase to have the images render properly in github pages. 
* clean up plugin was configured to remove the duplicated wiki/resources/ directory

To test locally run "mvn site" and then "mvn site:run"

### Benefits
Maven handles site generation based on the existing markdown files in the repository

### Possible Drawbacks
User have to run maven commands locally before push to generate docs/ directory (and also pushing the generated files, causing a bigger pull request). Investigation of how to automate this in a travis build step is ongoing, but for now I opened this pull request to be reviewed.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Emelie Pettersson emelie.pettersson@ericsson.com

### CODE REVIEW

File: wiki/markdown/Add-Subscription.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+select a specific template for the subscription form. Currently, the following 
+options are listed: Jenkins Pipeline Parametrized Job Trigger, REST POST and 
+Mail Trigger.
+
+**SubscriptionName:** A field to give a name to the current subscription. Only 
+letters, underscore and numbers are allowed in this field.
+
+**NotificationType:** A dropdown list with two options, i.e., REST POST and 
+Mail Trigger, depending on how a subscriber want to be notified when a 
+subscription is fulfilled.
```
Comment (a-breid-ericsson) [inline wiki/markdown/Add-Subscription.md]: Not really correct. It is radio buttons.

File: wiki/markdown/Add-Subscription.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+**RestPostMediaType:** A dropdown list that provides options for selecting a 
+specific form content type. The options available in the list depends on the 
+selected template type.
```
Comment (a-breid-ericsson) [inline wiki/markdown/Add-Subscription.md]: Also radio buttons

File: wiki/markdown/Add-Subscription.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+**NotificationMessage:** It is used to send a message to a specific client. The 
+format of the message depends on the options selected in the NotificationMessage 
+and RestPostMediaType. A button, “Add Key/Value pair”, in this field may be 
+used to add more messages.
+
+**Authorization:** A list to select authorization type. Currently, only one 
+authorization type is provided which is “BASIC_AUTH”. The option “NO_AUTH” 
+implies that authorization is not required.
+
+**Repeat:** A drop down list to choose either true or false. It depends whether
```
Comment (a-breid-ericsson) [inline wiki/markdown/Add-Subscription.md]: Slider.

File: wiki/markdown/Add-Subscription.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+**Authorization:** A list to select authorization type. Currently, only one 
+authorization type is provided which is “BASIC_AUTH”. The option “NO_AUTH” 
+implies that authorization is not required.
+
+**Repeat:** A drop down list to choose either true or false. It depends whether 
+same subscription should be re-triggered for new additions to the aggregated 
+object. If false, the notification will be triggered only the first time when 
+conditions are fulfilled. 
+
+**NotificationMeta:** A unique address need to be added for notification.
```
Comment (a-breid-ericsson) [inline wiki/markdown/Add-Subscription.md]: Moved to be above Authorization

File: wiki/markdown/Add-Subscription.md (hunk 1 above)
Comment (a-breid-ericsson) [inline wiki/markdown/Add-Subscription.md]: If Notification type MAIL is selected an email subject may be entered, but it is optional.

File: README.md [hunk 5]
```diff
@@ -36,13 +36,24 @@ The contents of this repository are licensed under the [Apache License 2.0](./LI
...
-1. [**GUI Overview**](./wiki/GUI-Overview.md)
-1. [**Subscription Handling**](./wiki/Subscription-Handling.md)
-    -  [**Add Subscription**](./wiki/Add-Subscription.md)
-1. [**Test Rules User Guide**](./wiki/Test-Rules.md)
+1. [**GUI Overview**](wiki/markdown/GUI-Overview.md)
+1. [**Subscription Handling**](wiki/markdown/Subscription-Handling.md)
+    -  [**Add Subscription**](wiki/markdown/Add-Subscription.md)
+1. [**Test Rules**](wiki/markdown/Test-Rules.md)
+    - [**Rule testing mechanism via "Test Rules" GUI interface**](wiki/markdown/Test-Rules.md#Rule-testing-mechanism-via-Test-Rules-GUI-interface)
+    - [**Introduction**](wiki/markdown/Test-Rules.md#Introduction)
```
Comment (a-breid-ericsson) [inline README.md]: Those sub headers was removed by a PR before this, should they really be added back?

File: wiki/markdown/Add-Subscription.md (hunk 1 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Add-Subscription.md]: Done

File: wiki/markdown/Add-Subscription.md (hunk 2 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Add-Subscription.md]: Done

File: wiki/markdown/Add-Subscription.md (hunk 4 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Add-Subscription.md]: Done

File: wiki/markdown/Add-Subscription.md (hunk 1 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Add-Subscription.md]: Done

File: README.md (hunk 5 above)
Comment (e-pettersson-ericsson) [reply README.md]: Fixed now

File: wiki/markdown/Subscription-Handling.md [hunk 6]
```diff
@@ -1,20 +1,30 @@
...
-This button indicates, through its colour, whether a backend instance is connected with frontend or not. The green colour means backend is connected while red means no instance is connected.
+This button indicates, through its colour, whether a backend instance is 
+connected with frontend or not. The green colour means backend is connected 
+while red means no instance is connected.
 #### _Search_
-To search subscriptions with matching names in the database (displayed in the table).
+To search subscriptions with matching names in the database (displayed in the 
+table).
 
 **_More Subscriptions related information can be found [here](https://github.com/eiffel-community/eiffel-intelligence/tree/master/wiki/Subscription-API.md) and [here](https://github.com/eiffel-community/eiffel-intelligence/tree/master/wiki/Subscriptions.md)_**
```
Comment (e-edling-ericsson) [inline wiki/markdown/Subscription-Handling.md]: These links will be changed when EI backend merges, make sure to keep them updated

File: wiki/markdown/Test-Rules.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+            ...
+        ] 
+    }
+    
+<p>And then run curl command below.</p>
+
+    curl -X POST -d "@<path to file>" -H "Content-Type: application/json" http://<host>:8080/rules/rule-check/aggregation?backendurl="http://127.0.0.1:8090/"
+
+---
+**_More information about how to write rules can be found [here](https://github.com/eiffel-community/eiffel-intelligence/blob/master/wiki/Rules.md)._**
```
Comment (e-edling-ericsson) [inline wiki/markdown/Test-Rules.md]: Remember to update this as well if the path is changed

File: wiki/markdown/Subscription-Handling.md (hunk 6 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Subscription-Handling.md]: Done

File: wiki/markdown/Test-Rules.md (hunk 7 above)
Comment (e-pettersson-ericsson) [reply wiki/markdown/Test-Rules.md]: Done

File: README.md [hunk 8]
```diff
@@ -43,6 +43,6 @@ Visit [Eiffel Community](https://eiffel-community.github.io) to get started and
 # Documentation
 
 1. [**GUI Overview**](./wiki/GUI-Overview.md)
-1. [**Subscription Handling**](./wiki/Subscription-Handling.md)
-    -  [**Add Subscription**](./wiki/Add-Subscription.md)
-1. [**Test Rules User Guide**](./wiki/Test-Rules.md)
+1. [**Subscription Handling**](./wiki/markdown/Subscription-Handling.md)
```
[the remark this answers is not in the document]
Comment (e-pettersson-ericsson) [reply README.md]: Faulty link. It is now fixed.

**Pull request merged**